package-lock.json

```diff
@@ -1,8 +1,5709 @@
 {
```

Comment:
What are the benefits for this change? It seems that `v2` is not totally backwards compatible with previous versions of NPM 🤔

Reply:
~~npm updated this file automatically when I run `npm install`.  If we decide to use Node.js 16 we should let its default behavior not produce any unstaged changes.~~

Good point. I tested with Node.js 14 and it will rollback to `lockfileVersion: 1`.

This change is not friendly with npm 6 (Node.js 12, 14) users.

Reply:
> Good point. I tested with Node.js 14 and it will rollback to `lockfileVersion: 1`.
>
> This change is not friendly with npm 6 (Node.js 12, 14) users.

Good thing I read this and tested it myself :sweat_smile:  For some reason [I thought npm 6 does not rollback to `lockfileVersion: 1`](https://github.com/simple-icons/simple-icons/pull/6179#discussion_r679402311) but as you said it does do that.

This complicates things a bit as it doesn't make the decision as clearcut as I'd hoped. Staying with `lockfileVersion: 1` is annoying for people with npm v7. Switching to `lockfileVersion: 2` is annoying for people with npm 6 (or lower). There's a couple of solution but none of them actually solve the problem. One is to use a `Dockerfile` for development, but given the low bar for entry we aim for I don't like that. A second is to use an `.nvmrc` file to specify the Node (and by extension NPM) version, which I don't like for the same reason. A third solution is to add  `"npm": ">=7.0.0"` to [the engines in `package.json`](https://github.com/simple-icons/simple-icons/blob/0ccb7cbfaf55b7151c5fbeec7075c53fbbc18a95/package.json#L49-L51), but that's only a solution if the contributes has set the `engine-strict` flag manually which, again, I don't think we should expect from our contributors.

Clearly, at some point we need to make the switch, given my experience reviewing Pull Requests for this project, a majority of contributors is still using NPM v6 (presumably because they installed Node quite some time ago and ignore the notice to update NPM) so given the above I actually think we should stick with `lockfileVersion: 1` until we see more Pull Requests that accidentally update `package-lock.json`.

In the meantime, we could implement all of the above in an attempt to make it easier on those contributors that use the relevant tools/configurations. @LitoMore (and @mondeja), what do you think? If you agree with my (new) opinion, do you want to implement it here or should I create a (new) Pull Request for that?

Reply:
@LitoMore, did you see this comment?

Reply:
@ericcornelissen Sorry I missed it.

Agrees. Let's stick with `lockfileVersion: 1`.

Reply:
Or just drop the lock file? Add `package-lock=false` to `.npmrc`. As [Sindre Sorhus does](https://github.com/sindresorhus/ama/issues/479#issuecomment-310661514).

Reply:
> I didn't find a way to stick `lockfileVersion: 1`. Any thought?

Not sure if I follow, seems like your PR doesn't change `package-lock.json` anymore - did you manage to resolve the problem yourself?

> Or just drop the lock file? Add `package-lock=false` to `.npmrc`

Interesting suggestion, didn't know about that. I'm not totally against it, but the reasons lined out in the thread you linked do not apply here. This project [only has `devDependencies`](https://github.com/simple-icons/simple-icons/blob/96c32669d2d450bc8c85c7b748b3bb9f2876c312/package.json#L20-L32) and I, as pointed out by others in the thread, a `package-lock.json` provides is with a deterministic dev environment.

What do you think about that @LitoMore?

Reply:
> Did you manage to resolve the problem yourself?

There are too many conflict in the file and hard to solve. We could handle it after determining the solution.

> This project only has devDependencies.

I think it's OK to drop lock file. Even we can face problems in the development environment earlier. If there is any dependency problem and hard to solve, just pin it.

Reply:
> There are too many conflicts in the file and hard to solve. We could handle it after determining the solution.

Sure

> I think it's OK to drop lock file. Even we can face problems in the development environment earlier. If there is any dependency problem and hard to solve, just pin it.

We can pin our `devDependency`s - which we already do - but not their dependencies, that's what the lockfile provides us with.

But again, I'm not against it. I'm definitely open to at least trying it out. In that case, could you add `.npmrc` file and revert all changes so that we're still on Node v12 and npm v6 in this Pull Request? I propose we take the discussion/work for upgrading to Node v16 in the CI, adding `.nvmrc` and updating the `Dockerfile` to a different Pull Request/issue - the sooner this is merged the sooner other contributors benefit from the change :slightly_smiling_face:

Reply:
Updated.

Reply:
I agree overall with the solution, but just for curiosity: how [renovate security updates](https://github.com/simple-icons/simple-icons/pull/6199) are handled after this? Are just ignored? Additionally, if we do this, we must remove [this part of the renovate configuration](https://github.com/simple-icons/simple-icons/blob/0a3621df0e5cb43a7a818e5c29cb07494ada5c39/.github/renovate.json5#L9-L20)?